framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -429,44 +412,43 @@ public void standardizeReturnType(
             return;
         }
 
-        ExecutableElement methodElt = TreeUtils.elementFromDeclaration(methodDeclTree);
-
-        standardizeForMethodSignature(
-                methodDeclTree, pathToMethodDecl, methodElt, atm, removeErroneousExpressions);
+        JavaExpressionContext context =
+                JavaExpressionContext.buildContextForMethodDeclaration(
+                        methodDeclTree, pathToMethodDecl, factory.getChecker());
+        standardizeDoNotUseLocalScope(context, pathToMethodDecl, atm);
     }
 
     /**
-     * Standardizes the Java expressions in annotations for a method signature location, including:
-     *
-     * <ul>
-     *   <li>type annotations on a return type, formal parameter type, or exception type, and
-     *   <li>declaration annotations on a method (such as a pre- or post-condition contract
-     *       annotation) or formal parameter.
-     * </ul>
+     * Viewpoint-adapt all dependent type annotations to the method declaration, {@code
+     * methodDeclTree}. This changes occurrences of formal parameter names to "#2" syntax, and it
+     * removes expressions that contain other local variables.
      *
-     * @param methodDeclTree a method declaration
-     * @param pathToMethodDecl the path to the method declaration
-     * @param elt the element for the method or a formal parameter; used for obtaining the enclosing
-     *     class
-     * @param atm a type that has a dependent type annotation; is side-effected by this method
-     * @param removeErroneousExpressions if true, remove erroneous expressions rather than
-     *     converting them into an explanation of why they are illegal
+     * @param methodDeclTree the method declaration to which the annotations are viewpoint-adapted
+     * @param atm type to viewpoint-adapt; is side-effected by this method
      */
-    public void standardizeForMethodSignature(
-            MethodTree methodDeclTree,
-            TreePath pathToMethodDecl,
-            Element elt,
-            AnnotatedTypeMirror atm,
-            boolean removeErroneousExpressions) {
-        if (!hasDependentAnnotations()) {
+    public void delocalize(MethodTree methodDeclTree, AnnotatedTypeMirror atm) {
+        if (!hasDependentType(atm)) {
             return;
         }
 
-        TypeMirror enclosingType = ElementUtils.enclosingTypeElement(elt).asType();
+        TreePath pathToMethodDecl = factory.getPath(methodDeclTree);
+        if (pathToMethodDecl == null) {
+            return;
+        }
+        // TODO: 1.) parameter names need to be coverted to the # index syntax.
```

Comment:
Do these TODOs mean that the method is not fully implemented?  (Maybe that will be done in a future pull request?)  Or should the TODO comments be removed and the documentation be updated?

Reply:
This method is not fully implemented. It will be implemented after other `useLocalScope` is removed.  I had to add this method now to simplify the code and make it easier to remove `useLocalScope`.